mod.js

```diff
@@ -0,0 +1,23 @@
+
+var generateString = (def_length=8, special = 1, digits = 1, uppercase = 1)=> {
+  var text = "";
+  var possible = `abcdefghijklmnopqrstuvwxyz`;
+  var specialChar = "!#$%&?"
+  for (var i = 0; i < def_length; i++){
```

Comment:
If I run your application with this command:
`node app.js -l 12 -d 5 -s 3 -u 4`
I'm getting wrong result:
`mHtwgsgtJJnm$?&70852`
As you see total password length was defined as 12 characters, but I'm getting password 20 characters long.
Problem is here, because you generate 12 letters and then add digits and special characters.